L1Trigger/L1TTrackMatch/plugins/L1TrackJetProducer.cc

```diff
@@ -198,194 +215,175 @@ void L1TrackJetProducer::produce(Event &iEvent, const EventSetup &iSetup) {
   }
+
+  std::sort(
+      L1L1TrackJetProducer->begin(), L1L1TrackJetProducer->end(), [](auto &a, auto &b) { return a.pt() > b.pt(); });
+  if (displaced_)
+    iEvent.put(std::move(L1L1TrackJetProducer), "L1TrackJetsExtended");
+  else
+    iEvent.put(std::move(L1L1TrackJetProducer), "L1TrackJets");
 }
 
 void L1TrackJetProducer::beginStream(StreamID) {}
 
 void L1TrackJetProducer::endStream() {}
 
-bool L1TrackJetProducer::trackQualityCuts(int trk_nstub, float trk_chi2, float trk_bendchi2) {
-  bool PassQuality = false;
-  if (!displaced_) {
-    if (trk_nstub == 4 && trk_bendchi2 < nStubs4PromptBend_ &&
-        trk_chi2 < nStubs4PromptChi2_)  // 4 stubs are the lowest track quality and have different cuts
-      PassQuality = true;
-    if (trk_nstub > 4 && trk_bendchi2 < nStubs5PromptBend_ &&
-        trk_chi2 < nStubs5PromptChi2_)  // above 4 stubs diffent selection imposed (genrally looser)
-      PassQuality = true;
-  } else {
-    if (trk_nstub == 4 && trk_bendchi2 < nStubs4DisplacedBend_ &&
-        trk_chi2 < nStubs4DisplacedChi2_)  // 4 stubs are the lowest track quality and have different cuts
-      PassQuality = true;
-    if (trk_nstub > 4 && trk_bendchi2 < nStubs5DisplacedBend_ &&
-        trk_chi2 < nStubs5DisplacedChi2_)  // above 4 stubs diffent selection imposed (genrally looser)
-      PassQuality = true;
-  }
-  return PassQuality;
-}
-
 void L1TrackJetProducer::fillDescriptions(ConfigurationDescriptions &descriptions) {
-  // l1tTrackJets
-  edm::ParameterSetDescription desc;
-  desc.add<edm::InputTag>("L1TrackInputTag", edm::InputTag("l1tTTTracksFromTrackletEmulation", "Level1TTTracks"));
-  desc.add<edm::InputTag>("L1PVertexCollection", edm::InputTag("l1tVertexProducer", "l1vertices"));
-  desc.add<double>("MaxDzTrackPV", 1.0);
-  desc.add<double>("trk_zMax", 15.0);
-  desc.add<double>("trk_ptMax", 200.0);
-  desc.add<double>("trk_ptMin", 3.0);
-  desc.add<double>("trk_etaMax", 2.4);
-  desc.add<double>("nStubs4PromptChi2", 5.0);
-  desc.add<double>("nStubs4PromptBend", 1.7);
-  desc.add<double>("nStubs5PromptChi2", 2.75);
-  desc.add<double>("nStubs5PromptBend", 3.5);
-  desc.add<int>("trk_nPSStubMin", -1);
-  desc.add<double>("minTrkJetpT", -1.0);
-  desc.add<int>("etaBins", 24);
-  desc.add<int>("phiBins", 27);
-  desc.add<int>("zBins", 1);
-  desc.add<double>("d0_cutNStubs4", -1);
-  desc.add<double>("d0_cutNStubs5", -1);
-  desc.add<int>("lowpTJetMinTrackMultiplicity", 2);
-  desc.add<double>("lowpTJetThreshold", 50.0);
-  desc.add<int>("highpTJetMinTrackMultiplicity", 3);
-  desc.add<double>("highpTJetThreshold", 100.0);
-  desc.add<bool>("displaced", false);
-  desc.add<double>("nStubs4DisplacedChi2", 5.0);
-  desc.add<double>("nStubs4DisplacedBend", 1.7);
-  desc.add<double>("nStubs5DisplacedChi2", 2.75);
-  desc.add<double>("nStubs5DisplacedBend", 3.5);
-  desc.add<int>("nDisplacedTracks", 2);
-  descriptions.add("l1tTrackJets", desc);
+  ParameterSetDescription desc;
+  desc.setUnknown();
```

Comment:
Please restore a true fillDescription method: why did you remove it?

Reply:
I restored it. and added to the emulation as well